exporter/datadogexporter/internal/translator/metrics_translator.go

```diff
@@ -33,40 +31,22 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/metrics"
 )
 
-type ttlCache struct {
-	cache *gocache.Cache
+// HostnameProvider gets a hostname
+type HostnameProvider interface {
+	// Hostname gets the hostname from the machine.
+	Hostname(ctx context.Context) (string, error)
 }
 
-// numberCounter keeps the value of a number
-// monotonic counter at a given point in time
-type numberCounter struct {
-	ts    uint64
-	value float64
+type Translator struct {
+	prevPts          *TTLCache
+	logger           *zap.Logger
+	cfg              config.MetricsConfig
+	buildInfo        component.BuildInfo
+	hostnameProvider HostnameProvider
```

Comment:
I think we should call that field `fallbackHostProvider`, which describes more precisely what that specific `HostnameProvider` is used for in the translator?